templates/s3/s3.py

```diff
@@ -699,16 +700,23 @@ def _monitor_minio(self):
             self.state.check('actions', 'install', 'ok')
         except StateCheckError:
             return
-        vm_robot, _ = self._vm_robot_and_ip()
-        minio = vm_robot.services.get(template_uid=MINIO_TEMPLATE_UID, name=self.guid)
-        state = minio.state
 
-        try:
-            disk_state = state.get('vm', 'disk')
-            self.state.set('vm', 'disk', disk_state['disk'])
-        except:
-            # probably no state set on the minio disk
-            pass
+        def get_state():
+            retries = 3
+            for _ in range(retries):
+                try:
+                    vm_robot, _ = self._vm_robot_and_ip()
+                    minio = vm_robot.services.get(template_uid=MINIO_TEMPLATE_UID, name=self.guid)
+                    state = minio.state
+                    return state
+
+                except HTTPError as e:
+                    if e.response.status_code == 500:
+                        gevent.sleep(10)
+
+            else:
+                self.state.delete('vm', 'disk')
```

Comment:
I think the caller should decide what to do (in this case delete state). But a `get` method should not have a side effect.